### ISSUE TITLE
This module has a direct dependency on angular core

### ISSUE THREAD

**Issue by codebanesr** [MRebati/angular7-pubsub#2]
this module depends on angular core and when we install it, it results in one extra angular core package and the compiler gives an error . Removing node_modules package removes the error and the library works as expected. Here is the error.

ERROR in node_modules/@angular/core/src/render3/ng_dev_mode.d.ts(9,11): error TS2451: Cannot redeclare block-scoped variable 'ngDevMode'.
node_modules/angular7-pubsub/node_modules/@angular/core/src/render3/ng_dev_mode.d.ts(9,11): error TS2451: Cannot redeclare block-scoped variable 'ngDevMode'.

**Comment by MRebati** [comment]
what's the version of angular you're on? can we have more details on the packages? a repro would be helpful

**Comment by codebanesr** [comment]
sure will do that. I am on angular 7 though.

Thanks,
Shanur

**Comment by MRebati** [comment]
I can not remove the angular/core dependency from project.
the only option available is to update the package dependencies to the latest versions on project.

if that's not urgent I can do in these couple days. if so please send a pull-request

**Comment by codebanesr** [comment]
**Use TypeScript path mapping for peer dependencies**
Angular libraries should list all @angular/* dependencies as peer dependencies. This ensures that when modules ask for Angular, they all get the exact same module. If a library lists @angular/core in dependencies instead of peerDependencies, it might get a different Angular module instead, which would cause your application to break. 


Add this to ./tsconfig.json
```
{
  "compilerOptions": {
    // ...
    // paths are relative to `baseUrl` path.
    "paths": {
      "@angular/*": [
        "./node_modules/@angular/*"
      ]
    }
  }
}
```

More on this [here](https://angular.io/guide/creating-libraries)

**Comment by MRebati** [comment]
merged to master #3
